src/visualizers/Histogram.ts

```diff
@@ -158,44 +159,43 @@ class FactorHistogramVisualizer extends VisualizerDefault {
     // Create a number that represents how
     // many pixels wide each bin should be
     binWidth(): number {
+        const width = this.sketch?.width ?? 0
         // 0.95 Creates a small offset from the side of the screen
         if (this.binFactorArray.length <= 30) {
-            return (0.95 * this.sketch.width) / this.binFactorArray.length - 1
+            return (0.95 * width) / this.binFactorArray.length - 1
         } else {
-            return (0.95 * this.sketch.width) / 30 - 1
+            return (0.95 * width) / 30 - 1
         }
     }
 
     // Create a number that represents how many pixels high
     // each increase of one in the bin array should be
     height(): number {
+        const height = this.sketch?.height ?? 0
         const greatestValue = this.binFactorArray.reduce(
             (a: number, b: number) => Math.max(a, b),
             -Infinity
         )
         // magic number creates a small offset from the top of the screen
-        return (0.9 * this.sketch.height) / greatestValue
+        return (0.9 * height) / greatestValue
     }
 
     // check if mouse is in the given bin
     mouseOverInBin(xAxisHeight: number, binIndex: number): boolean {
-        if (
-            this.sketch.mouseY
-                // hard to mouseover tiny bars; min height to catch mouse
+        const y = this.sketch?.mouseY ?? 0
+        // hard to mouseover tiny bars; min height to catch mouse
+        return (
+            y
                 > Math.min(
                     xAxisHeight
                         - this.height() * this.binFactorArray[binIndex],
                     xAxisHeight - 10
-                )
-            // and above axis
-            && this.sketch.mouseY < xAxisHeight
-        ) {
-            return true
-        }
-        return false
+                ) && y < xAxisHeight
+        ) // and above axis
     }
 
     drawHoverBox(binIndex: number, offset: number) {
+        if (!this.sketch) return
```

Comment:
please explain -- maybe you already did above

Reply:
Yup, I think I've covered the bases on this now, let me know your thoughts.